projects/self-creation-portal/src/app/components/resource-holder/resource-holder.component.ts

```diff
@@ -34,163 +34,93 @@ export class ResourceHolderComponent implements OnInit{
-    },
-    {
-      "id": 4,
-      "title": "sample project",
-      "type": "project",
-      "organization": {
-        "id": 24,
-        "name": "Tunerlabs",
-        "code": "tl"
-      },
-      "creator_name": "Suma",
-      "status": "DRAFT",
-      "actionButton": [{ action: 'EDIT', label: 'Edit', background_color: 'primary' }, { action: 'DELETE', label: 'Delete', background_color: 'warn' }]
-    },
-    {
-      "id": 5,
-      "title": "sample project",
-      "type": "project",
-      "organization": {
-        "id": 24,
-        "name": "Tunerlabs",
-        "code": "tl"
-      },
-      "creator_name": "Suma",
-      "status": "DRAFT",
-      "actionButton": [{ action: 'EDIT', label: 'Edit', background_color: 'primary' }, { action: 'DELETE', label: 'Delete', background_color: 'warn' }]
-    },
-    {
-      "id": 6,
-      "title": "sample project",
-      "type": "observation with rubrics",
-      "organization": {
-        "id": 24,
-        "name": "Tunerlabs",
-        "code": "tl"
-      },
-      "creator_name": "Suma",
-      "status": "DRAFT",
-      "actionButton": [{ action: 'EDIT', label: 'Edit', background_color: 'primary' }, { action: 'DELETE', label: 'Delete', background_color: 'warn' }]
-    },
-    {
-      "id": 7,
-      "title": "sample project",
-      "type": "observation",
-      "organization": {
-        "id": 24,
-        "name": "Tunerlabs",
-        "code": "tl"
-      },
-      "creator_name": "Suma",
-      "status": "DRAFT",
-      "actionButton": [{ action: 'EDIT', label: 'Edit', background_color: 'primary' }, { action: 'DELETE', label: 'Delete', background_color: 'warn' }]
-    }
-  ]
-
+  lists:any = []
   constructor(private route: ActivatedRoute, private formService: FormService) {
   }
 
   ngOnInit() {
     this.loadSidenavData();
-    this.filters.filteredLists = this.lists;
-    this.pagination.totalCount = this.lists.length;
-    this.updateCurrentList();
+    this.getList();
   }
 
   loadSidenavData(){
     const currentUrl = this.route.snapshot.routeConfig?.path;
+    const isDraftsUrl = !!currentUrl && currentUrl.includes('drafts');
     this.formService.getForm(SIDE_NAV_DATA).subscribe(form => {
       const currentData = form?.result?.data.fields.controls.find((item: any) => item.url === currentUrl)?.filterData;
       this.filters.filterData = currentData || [];
       this.filters.showChangesButton = this.filters.filterData.some((filter: any) => filter.label === 'Status');
+      this.filters.showActionButton = isDraftsUrl;
     });
   }
 
   onPageChange(event: any) {
     this.pagination.pageSize = event.pageSize;
     this.pagination.currentPage = event.page - 1;
-    this.updateCurrentList();
+    this.getList(); 
   }
 
   receiveSearchResults(event: string) {
     this.filters.search = event.trim().toLowerCase();
     this.pagination.currentPage = 0;
-    this.applyFiltersAndSearch();
     this.paginationComponent.resetToFirstPage();
   }
 
   onFilterChange(event: any) {
```

Comment:
why filtering and sorting functions in resource-holder component ? move all the logics related to filter and sort or search related to filter page component